assets/app/scripts/services/docker.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("DockerImageGeneratorService", function($filter, ApplicationGenerator, DataService) {
+    var findImage = function(name, projectContext) {
+      var importImage = {
+        kind: "ImageStreamImport",
+        apiVersion: "v1",
+        metadata: {
+          name: "newapp",
```

Comment:
add a comment that this is just a placeholder name because name is required even when import is false